### PR TITLE
feat: custom SSL CA support for SQL data sources

### PR DESCRIPTION
**Pull request by palpatim**
#### Description of changes:

Add documentation for specifying custom SSL CA certificate bundle for SQL Lambda data sources.

image

#### Related GitHub issue #, if available:

- https://github.com/aws-amplify/amplify-backend/pull/1696

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

This should not be submitted until https://github.com/aws-amplify/amplify-backend/pull/1696 is released.

Which product(s) are affected by this PR (if applicable)?
- [x] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [x] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] Swift
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/[platform]/build-a-backend/data/connect-to-existing-data-sources/connect-postgres-mysql-database/index.mdx [hunk 1]
```diff
@@ -135,6 +127,20 @@ However, there is a tradeoff of increased latency - queries may take slightly lo
 
 </Accordion>
 
+<Accordion title="Connecting to a database with a custom SSL certificate">
+
+Amplify creates an [AWS Lambda](https://aws.amazon.com/lambda) function using a Node.js runtime to connect your AppSync API to your SQL database. The Lambda function connects to the database using Secure Socket Layer (SSL) or Transport Layer Security (TLS) to protect data in transit. Amplify automatically uses the correct root certificate authority (CA) certificates for Amazon RDS databases, and the Node.js runtime includes root CAs from [well-known certificate providers](https://github.com/nodejs/node/issues/4175) to connect to non-RDS databases.
+
+However, if your database uses a custom or self-signed SSL certificate, you can upload the PEM-encoded public CA certificate to your Amplify project as an Amplify Secret when you generate the database configuration, and specify that secret when generating the schema from your database:
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/connect-to-existing-data-sources/connect-postgres-mysql-database/index.mdx:134]:
```suggestion
However, if your database uses a custom or self-signed SSL certificate, you can upload the PEM-encoded public CA certificate to your Amplify project as a secret when you generate the database configuration, and specify that secret when generating the schema from your database:
```

removed branded name for consistency with other references

https://docs.amplify.aws/react/deploy-and-host/fullstack-branching/secrets-and-vars/
https://docs.amplify.aws/react/build-a-backend/functions/environment-variables-and-secrets/#secrets

File: src/pages/[platform]/build-a-backend/data/connect-to-existing-data-sources/connect-postgres-mysql-database/index.mdx [hunk 2]
```diff
@@ -135,6 +127,20 @@ However, there is a tradeoff of increased latency - queries may take slightly lo
...
+
+```
```
Comment (josefaidt) [inline src/pages/[platform]/build-a-backend/data/connect-to-existing-data-sources/connect-postgres-mysql-database/index.mdx:136]:
````suggestion
```bash title="Terminal" showLineNumbers={false}
````

added flashy snippet title

File: src/pages/[platform]/build-a-backend/data/connect-to-existing-data-sources/connect-postgres-mysql-database/index.mdx [hunk 3]
```diff
@@ -135,6 +127,20 @@ However, there is a tradeoff of increased latency - queries may take slightly lo
...
+
+However, if your database uses a custom or self-signed SSL certificate, you can upload the PEM-encoded public CA certificate to your Amplify project as a secret when you generate the database configuration, and specify that secret when generating the schema from your database:
+
+```bash title="Terminal" showLineNumbers={false}
+npx ampx sandbox secret set CUSTOM_SSL_CERT < /path/to/custom/ssl/public-ca-cert.pem
+npx ampx generate schema-from-database --connection-uri-secret SQL_CONNECTION_STRING --ssl-cert-secret CUSTOM_SSL_CERT --out amplify/data/schema.sql.ts
+```
+
+The Lambda function will then use the specified root CA to validate connections to the database.
+
```
Comment (sundersc) [inline src/pages/[platform]/build-a-backend/data/connect-to-existing-data-sources/connect-postgres-mysql-database/index.mdx:142]: Should we callout that the size of the file must be <= 4 KB?

**Review by sundersc — APPROVED** [review]

**Review by hbuchel — COMMENTED** [review]
If you merge in latest from main, it includes a fix for the accessibility violations on this page which will unblock the checks.

**Comment by palpatim** [comment]
Force-pushed a rebase from `main`

**Review by sundersc — APPROVED** [review]

**Review by josefaidt — APPROVED** [review]

**Review by chrisbonifacio — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

**Pull request merged**